src/Multivariate/ProductFun.jl

```diff
@@ -241,7 +243,6 @@ function coefficients(f::ProductFun,ox::Space,oy::Space)
 end
 
 (f::ProductFun)(x,y) = evaluate(f,x,y)
-(f::ProductFun)(x,y,z) = evaluate(f,x,y,z)
```

Comment:
If I understand correctly, the new type `TensorIteratorFun` will handle nD functions efficiently. However, do we need to remove `ProductFun` methods?

Reply:
No. Actually, the `TensorIteratorFun` does not work for all bases, only for bases which have infinite coefficients, which are all except `ConstantSpace`, `SubSpace`, and some others. These spaces are referred to in the code as Trivial spaces, and hence `TrivialTensorizer` in TensorSpace.jl. The creation of a tensor for these kind of spaces is just simpler than for others and more efficient.
I will add a check in the constructor of `TensorIteratorFun` that the spaces are really only of this type.

`ProductFun` also handles "nonTrivialSpaces" but only works in 2D. I therefore deleted this method for 3D, the program currently just hangs when trying to evaluate this function. 

The reason `ProductFun` is faster for 2D polynomials than the `TensorIteratorFun`, is that for the 2D case there is a simpler method than Stars and Bars to convert the coefficient vector to a tensor. But `ProductFun` actually builds the full tensor, which is not efficient in the high dimensional setting. Therefore, in `TensorIteratorFun` I never actually build the full tensor but calculate the result directly from the `Tensorizer` iterator.

Also, currently the "detensorizer" for the nD case is missing. For this reason some things working in 2D will not work in 3D (I think multiplication and some others, but addition works).
But I think adding this is maybe better for another PR, since this already contains a lot of changes and it is not breaking with anything, since this also did not work before.

Reply:
It'll be better if you comment the code out instead of deleting it, and leave this explanation as a comment